go/kbnm/handler.go

```diff
@@ -24,8 +25,26 @@ func handleChat(req *Request) error {
 	}
 
 	// FIXME: Get the absolute path without a filled PATH var somehow?
-	cmd := exec.Command("/usr/local/bin/keybase", "chat", "send", "--private", req.To, req.Body)
+	cmd := exec.Command("/usr/local/bin/keybase", "chat", "send", "--private", req.To)
 
-	// TODO: Check/convert status code more precisely?
-	return cmd.Run()
+	// Write message body over STDIN to avoid running up against bugs with
+	// super long messages.
+	stdin, err := cmd.StdinPipe()
+	if err != nil {
+		return err
+	}
+	defer stdin.Close()
+
+	if err := cmd.Start(); err != nil {
+		return err
+	}
+
+	if _, err = io.WriteString(stdin, req.Body); err != nil {
+		return err
+	}
+	stdin.Close()
+
+	// TODO: Check/convert status code more precisely? Maybe return stdout as
+	// part of the error if there is one?
+	return cmd.Wait()
```

Comment:
Eh, I'm starting to suspect your goroutine version might save is some pain :p

I think the Most Correct Unix Thing here would be to avoid short-circuiting on those errors. So assuming `Start` succeeds, you always `WriteString`, `Close`, and `Wait` in that order, even if one of the first two return errors. Then at the end, if you did get errors, you return one of them with whatever precedence rule makes sense to you.